ci/travis_script_python.sh

```diff
@@ -4,28 +4,19 @@ set -e
 
 PYTHON_DIR=$TRAVIS_BUILD_DIR/python
 
+# Re-use conda installation from C++
+export MINICONDA=$TRAVIS_BUILD_DIR/miniconda
+export PATH="$MINICONDA/bin:$PATH"
+export LD_LIBRARY_PATH="$MINICONDA/lib:$LD_LIBRARY_PATH"
```

Comment:
I think the issue is that parquet-cpp is not being installed in the conda environment where pyarrow library is being built and tests run (https://github.com/apache/arrow/blob/master/ci/travis_script_python.sh#L32). 

Two related issues, then: 

1) parquet-cpp needs to be added to the build and runtime requirements in python/conda.recipe/meta.yaml

2) parquet-cpp must be installed after `conda create` is installed -- it's only being installed in the primary / top level environment

I think if you do this you can get rid of the LD_LIBRARY_PATH hack